src/plots/cartesian/axes.js

```diff
@@ -389,6 +389,10 @@ axes.saveRangeInitial = function(gd, overwrite) {
             ax._rangeInitial = ax.range.slice();
             hasOneAxisChanged = true;
         }
+        // store the initial range for the rangeslider if we zoom on oppaxis
+        if((isNew && ax.fixedrange === false) || (overwrite && hasChanged)) {
+            ax._rangesliderInitialRange = ax.range.slice();
```

Comment:
Hmm, one issue with this: `_rangesliderInitialRange` isn't part of the figure JSON... so if you zoom in on the y axis, then you decide you like the plot and you save it, or you snapshot it even, the rangeslider y axis will not have the same range in the snapshot or when you reopen the plot.

That seems to me to mean that we need a real attribute for the range of each y axis on the rangeslider, and perhaps even an attribute for whether to autorange that y axis on the range slider - it would be cool if the y axis could be told to autorange in the range slider even if you start out zoomed in on the main plot, wouldn't it?

BTW I came to this realization as I was thinking "what tests should we have for this feature", noting that a big part of this is drawing code so it should have an image test, but then realizing as it stands we *can't* make an image test because we have no mechanism to force `_rangesliderInitialRange` to be different from the y axis range in an image test, therefore we can't make the opp axis masks appear there.

Reply:
Right, I didn't think about that.
I'll look into it.